### ISSUE TITLE
Odin test always fails on latest

### ISSUE THREAD

**Issue by zhibog** [odin-lang/Odin#1223]
## Context

`odin test` fails, when it should succeed.

Win10 LTSC
`odin version dev-2021-10:3337d386`

## Expected Behavior

The test should succeed

## Current Behavior

It fails

## Failure Information (for bugs)

It works fine when tried with commit [49c761d](https://github.com/odin-lang/Odin/tree/49c761dc6db889301450483ff79842fa192b42a8)

### Steps to Reproduce

Get the latest commit and run the follwing via `odin test`.
Then run the same with the commit mentioned above.
Note: The condition does not really matter at all. It will always fail, regardless if the test proc does anything or not.

```odin
package main

import "core:testing"

@(test)
foo :: proc(t: ^testing.T) {
    s1 := "Hellope"
    s2 := "Hellope"

    if s1 != s2 {
        testing.error(t, "Strings should be equal")
    }
}
```

### Failure Logs

Ran with [49c761d](https://github.com/odin-lang/Odin/tree/49c761dc6db889301450483ff79842fa192b42a8)
```
[Package: main]
[Test: foo]
[foo : SUCCESS]
----------------------------------------
1/1 SUCCESSFUL
```

Ran with latest. Please note: the print does not occur, which is weird too.
```
[Package: main]
[Test: foo]
[foo : FAILURE]
----------------------------------------
0/1 SUCCESSFUL
```

**Comment by godindustry** [comment]
49c761d

**Comment by judah-caruso** [comment]
This is still the case after running the example code given on `dev-2022-04:b4f8efcb`

```
        Odin: dev-2022-04:b4f8efcb
        OS:   Windows 11 Professional (version: 21H2), build 22000.613
        CPU:  AMD Ryzen 9 5900HS with Radeon Graphics
        RAM:  40352 MiB
```

**Comment by totoki-kei** [comment]
I think this issue may be fixed by this pull-request's change.
https://github.com/odin-lang/Odin/pull/1829

**Comment by zhibog** [comment]
Fixed in [#1829](https://github.com/odin-lang/Odin/pull/1829)

Thank you very much
